lib/gui/modules/drive-scanner.js

```diff
@@ -110,9 +95,16 @@ const scan = () => {
       includeSystemDrives: settings.get('unsafeMode')
     },
     usbboot: {
-      files: {
-        'bootcode.bin': USBBOOT_BOOTCODE_BIN_BUFFER,
-        'start.elf': USBBOOT_START_ELF_BUFFER
+      readFile: (name) => {
+        const blobPath = _.attempt(() => {
+          if (name === 'bootcode.bin' || name === 'start_cd.elf' || 'fixup_cd.dat') {
+            return path.join('raspberrypi', name)
+          }
+
+          return name
+        })
+
+        return fs.readFileAsync(path.join(BLOBS_DIRECTORY, 'usbboot', blobPath))
```

Comment:
I can't quite tell, but does this cache the files, so that a second request for the same filename (e.g. config.txt) doesn't result in multiple fs.readFileAsync calls?

Reply:
Good call. Done!